src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs

```diff
@@ -192,13 +166,17 @@ protected override void AddAdornmentsToAdornmentLayer_CallOnlyOnUIThread(Normali
                     continue;
                 }
 
+                // This is what places the diagnostic UI at the end of the line of code or at the end of the TextView window.
                 Canvas.SetLeft(visualElement,
-                    tag.Location == InlineDiagnosticsLocations.PlacedAtEndOfCode ? lineView.Right :
-                    tag.Location == InlineDiagnosticsLocations.PlacedAtEndOfEditor ? TextView.ViewportRight - visualElement.DesiredSize.Width :
-                    throw ExceptionUtilities.UnexpectedValue(tag.Location));
+                    tag.Location == InlineDiagnosticsLocations.PlacedAtEndOfCode
+                        ? Math.Max(lineView.Right, lineView.LineTransform.Right)
```

Comment:
get either the rightmost location of the line, or the LineTransform's rightmost location which may include other adornments.